### ISSUE TITLE
Issue on page /examples/02_all_pass_filter.html

### ISSUE THREAD

**Issue by philippevelha** [gdsfactory/sax#37]
I defined an add-drop like:

    
ring, _ = sax.circuit(
    netlist={
        "instances": {
            "lft": coupler,
            "top": waveguide,
            "bot": waveguide,
            "rgt": coupler,
        },
        "connections": {
            "lft,in1": "top,in0",
            "lft,out1":"bot,in0",
            "rgt,in1": "top,out0",
            "rgt,out1":"bot,out0",
        },
        "ports": {
            "in0": "lft,in0",
            "in1": "rgt,in1",
            "out0": "lft,out0",
            "out1": "rgt,out1",
        },
    }
)

I simulate the behaviour like below:

wl = jnp.linspace(1.54, 1.56, 10001)
result = ring(wl=wl, lft={'coupling': 0.02}, top={'length': 100.0,'loss':0.00155}, bot={'length': 100.0,'loss':0.00155}, rgt={'coupling': 0.02})
print(result.values().__sizeof__())
plt.plot(1e3*wl, jnp.abs(result['in0', 'out0'])**2, label="in0->out0")
plt.plot(1e3*wl, jnp.abs(result['in0', 'in1'])**2, label="in0->in1", ls="--")
plt.plot(1e3*wl, jnp.abs(result['out0', 'out1'])**2, label="out0->out1", ls="--")
plt.xlabel("λ [nm]")
plt.ylabel("T")
plt.grid(True)
plt.figlegend(ncol=2, loc="upper center")
plt.show()

Whatever the backend I obtain a transmission greater than 1 or 2

**Comment by flaport** [comment]
I visualized the netlist with `saxviz.visualize_netlist` and got this:

![image](https://github.com/flaport/sax/assets/15832886/40394b6e-4589-413a-91f7-238a54867a3c)

clearly the ports on coupler `rgt` are doubly connected.

you can fix it like this:


```python
netlist={
    "instances": {
        "lft": "coupler",
        "top": "straight",
        "bot": "straight",
        "rgt": "coupler",
    },
    "connections": {
        "lft,in1": "top,in0",
        "lft,out1": "bot,in0",
        "rgt,in1": "top,out0",
        "rgt,out1": "bot,out0",
    },
    "ports": {
        "in0": "lft,in0",
        "in1": "rgt,in0",
        "out0": "lft,out0",
        "out1": "rgt,out0",
    },
}

models = {
    "straight": sax.models.straight,
    "coupler": sax.models.coupler,
}


ring, _ = sax.circuit(
    netlist=netlist,
    models =models
)
```

which gives the following netlist visualization:

![image](https://github.com/flaport/sax/assets/15832886/4a8eb2ec-e4aa-4c3d-9257-f021b0c33b1b)

And the following result:

![image](https://github.com/flaport/sax/assets/15832886/b4c6ad95-2a2e-45a5-90ee-f1fb589fbd6c)
